packages/cli/src/template/model.ts.ejs

```diff
@@ -31,21 +37,21 @@ export class <%= props.className %> implements Entity {
     }
 
     async save(): Promise<void> {
-        let id = this.id;
+        const id = this.id;
         assert(id !== null, "Cannot save <%=props.className %> entity without an ID");
-        await store.set('<%=props.entityName %>', id.toString(), this);
+        await store.set('<%=props.entityName %>', id.toString(), this as Compat<%=props.className %>Props);
```

Comment:
The type issue of the id field still needs to be addressed here.

` {...this, id: id.toString() } as CompatTransferProps`

Reply:
Functionally this is fine, its just the types we're getting around. Its better to not copy the data if we can avoid it